### DISCUSSION TITLE
При конкатенации превращает строку вида "-01" в число 1, что делать?

### DISCUSSION THREAD

**Discussion by aityomin** [totumonline/totum-ru-issues-and-discussions#196] [Неопределенное]
При конкатенации превращает строку вида "-01" в число 1, что делать?

**Comment by ttm-alexeypolunin** [comment]
Использовать тип поля Строка 🤷‍♀️:

<img width="440" alt="Снимок экрана 2022-08-17 в 12 49 54" src="https://user-images.githubusercontent.com/55755565/185089579-335a2c74-6c03-41e2-880e-7ef1a77602fb.png">

<img width="472" alt="Снимок экрана 2022-08-17 в 12 52 56" src="https://user-images.githubusercontent.com/55755565/185090112-529d8d07-f2e8-4667-aa79-4dd26f1640a9.png">

**Вот так не конкатенация, а математическое действие:**

```
= : "100" + "-01"

// result 99
```

**Reply by aityomin** [reply]
А так не должно срабатывать?

```
= : strAdd(str: $kod_programmy; str: $year; str:#nomer_povtora_v_godu)
```

**Reply by ttm-alexeypolunin** [reply]
<img width="900" alt="Снимок экрана 2022-08-18 в 11 24 44" src="https://user-images.githubusercontent.com/55755565/185347505-56628446-5d62-4857-aeef-79cf6007b815.png">

**Comment by aityomin** [comment]
Код Индекс:

```
= : strAdd(str: $kod_programmy; str: $year; str: #nomer_povtora_v_godu)

year: nowDate(format: "y")

kod_programmy: select(table: 'programmy'; field: 'kod_programmy'; where: 'programma' = #programmy)
```

![Снимок экрана (431)](https://user-images.githubusercontent.com/50951455/185369292-a3c05842-bf49-4865-87c7-908b0aa9be9b.png)

**Reply by ttm-alexeypolunin** [reply]
У вас `номер_повторного` — тип Селект.

`-01` это его отображение, а значение там наверное `1`.

Посмотрите эту часть учебного курса: https://ru.docs.totum.online/how-do-I-take-the-value-of-a-select

**Reply by aityomin** [reply]
Нет... Там значение из справочника...
![Снимок экрана (432)](https://user-images.githubusercontent.com/50951455/185371668-6cf80a9d-ed1a-4fd0-9436-e76a32ec6e26.png)

**Reply by ttm-alexeypolunin** [reply]
Да, подтверждаю — `-01` это его отображение, а значение там `1`. Смотрите видео: https://ru.docs.totum.online/how-do-I-take-the-value-of-a-select
